core/src/main/java/org/geysermc/geyser/translator/text/MessageTranslator.java

```diff
@@ -387,6 +387,38 @@ public static boolean isTooLong(String message, GeyserSession session) {
         return false;
     }
 
+    /**
+     * Normalizes whitespaces - a thing a vanilla client apparently does with commands and chat messages.
+     */
+    public static String normalizeSpace(String string) {
+        if (string == null || string.isEmpty()) {
+            return string;
+        }
+        final int size = string.length();
+        final char[] newChars = new char[size];
+        int count = 0;
+        int whitespacesCount = 0;
+        boolean startWhitespaces = true;
+        for (int i = 0; i < size; i++) {
+            final char actualChar = string.charAt(i);
+            final boolean isWhitespace = Character.isWhitespace(actualChar);
+            if (isWhitespace) {
+                if (whitespacesCount == 0 && !startWhitespaces) {
+                    newChars[count++] = ' ';
+                }
+                whitespacesCount++;
+            } else {
+                startWhitespaces = false;
+                newChars[count++] = (actualChar == 160 ? 32 : actualChar);
```

Comment:
Make these the char constants instead of their numeric counterparts.

Reply:
How is this? https://github.com/GeyserMC/Geyser/pull/4703/commits/04d902466401400587260ab0c182ccfab7c65614#diff-96a9adb1383acb0bfa5c957614f861bf2732c98886486a30d1ce13c7783dfe37R415

Reply:
Can you not do `' '`, etc? Or find the `\n` equivalent.

Reply:
Sure! Done here: https://github.com/GeyserMC/Geyser/pull/4703/commits/9b1fa74c9752245713a050a7d291ebf8fa3818e7